pkg/controllers/statusaggregator/plugins/deployment.go

```diff
@@ -103,6 +135,12 @@ func (receiver *DeploymentPlugin) AggregateStatues(
 		}
 	}
 
+	if needUpdateObservedGeneration {
```

Comment:
Let's add a comment to describe when `needUpdateObservedGeneration` will be true, and why we only update the observed generation when it is.

> We only update the source object's observed generation after it has been federated and synced, and we have aggregated the statuses of the latest cluster objects. It is only at this point where we can "successfully observe" the latest generation of the source object.